Nio/Authentication/LoginView.swift

```diff
@@ -36,6 +37,23 @@ struct LoginContainerView: View {
 
         store.login(username: username, password: password, homeserver: homeserverURL)
     }
+    
+    private func guessHomeserverURL() {
+        if !username.isEmpty && homeserver.isEmpty {
+            let userparts = username.split(separator: ":")
+            if userparts.count < 2 {
+                return
+            }
+            let mxautodiscovery = MXAutoDiscovery.init(domain: String(userparts[1]))
+            mxautodiscovery?.findClientConfig({(c) in
+                if self.homeserver.isEmpty { // Check again to prevent race condition.
+                    if c.wellKnown != nil {
+                        self.homeserver = c.wellKnown!.homeServer.baseUrl
+                    }
+                }
+            }, failure: {(e) in })
```

Comment:
Swiftlint is complaining about the short variable names for me, also we can safely unwrap `wellKnown`.
```suggestion
            mxautodiscovery?.findClientConfig({ config in
                // Check again to prevent race condition.
                if self.homeserver.isEmpty {
                    if let wellKnown = config.wellKnown {
                        self.homeserver = wellKnown.homeServer.baseUrl
                    }
                }
            }, failure: { _ in })
```